src/pycram/datastructures/dataclasses.py

```diff
@@ -99,53 +108,150 @@ class AxisAlignedBoundingBox:
         """
-        :return: The axis-aligned bounding box as a tuple of minimum and maximum points
+        :return: The minimum and maximum points of the bounding box
         """
         return self.get_min_point(), self.get_max_point()
 
     def get_min_point(self) -> Point:
         """
-        :return: The axis-aligned bounding box as a minimum point
+        :return: The minimum point of the bounding box
         """
         return Point(self.min_x, self.min_y, self.min_z)
 
     def get_max_point(self) -> Point:
         """
-        :return: The axis-aligned bounding box as a maximum point
+        :return: The maximum point of the bounding box
         """
         return Point(self.max_x, self.max_y, self.max_z)
 
     def get_min_max(self) -> Tuple[List[float], List[float]]:
         """
-        :return: The axis-aligned bounding box as a tuple of minimum and maximum points
+        :return: The minimum and maximum points of the bounding box as lists of floats
         """
         return self.get_min(), self.get_max()
 
     def get_min(self) -> List[float]:
         """
-        :return: The minimum point of the axis-aligned bounding box
+        :return: The minimum point of the axis-aligned bounding box as a list of floats
         """
         return [self.min_x, self.min_y, self.min_z]
 
     def get_max(self) -> List[float]:
         """
-        :return: The maximum point of the axis-aligned bounding box
+        :return: The maximum point of the axis-aligned bounding box as a list of floats
         """
         return [self.max_x, self.max_y, self.max_z]
 
 
+@dataclass
+class AxisAlignedBoundingBox(BoundingBox):
+
+    def get_corners(self) -> List[List[float]]:
+        """
+        :return: The points of the axis-aligned bounding box.
+        """
+        return self.get_axis_aligned_corners()
+
+    def get_transformed_box(self, transform: Transform) -> 'AxisAlignedBoundingBox':
+        """
+        Apply a transformation to the axis-aligned bounding box and return the transformed axis-aligned bounding box.
+
+        :param transform: The transformation to apply
+        :return: The transformed axis-aligned bounding box
+        """
+        transformed_points = transform.apply_transform_to_array_of_points(np.array(self.get_min_max()))
+        min_p = [min(transformed_points[:, i]) for i in range(3)]
+        max_p = [max(transformed_points[:, i]) for i in range(3)]
+        return AxisAlignedBoundingBox.from_min_max(min_p, max_p)
+
+    @classmethod
+    def from_min_max(cls, min_point: Sequence[float], max_point: Sequence[float]):
+        """
+        Set the axis-aligned bounding box from a minimum and maximum point.
+
+        :param min_point: The minimum point
+        :param max_point: The maximum point
+        """
+        return cls(min_point[0], min_point[1], min_point[2], max_point[0], max_point[1], max_point[2])
+
+
+@dataclass
+class RotatedBoundingBox(BoundingBox):
```

Comment:
Why dont use a polytope here

Reply:
We can add the convex hull version which is available directly from trimesh, but this does not remove the need for a rotated shape, because the rotation here is meant to rotate the shape to the current pose of the link, this rotation will also be needed in a polytope or a convex hull as we only get the points relative to the object origin and are aligned with world frame not with object current frame.

Reply:
I have added a method to get convex hull